### PR TITLE
fix: support web storage disabled browsers

### LINKED ISSUE #1773: Website does not work without localStorage/cookie access
Blocking cookies in browsers breaks https://deno.land in Firefox and Chrome due to localStorage being required:

Screenshot from 2021-06-12 15-29-35

```
DOMException: Failed to read the 'localStorage' property from 'Window': Access is denied for this document.
    at y.o [as constructor] (https://deno.land/_next/static/chunks/1f04d2384143d8e32f3ec5f902aa144732b686e1.7c7db78697ee6461a2ef.js:1:24890)
    at y.D [as render] (https://deno.land/_next/static/chunks/framework.da977932457ceef00f79.js:1:9130)
    at j (https://deno.land/_next/static/chunks/framework.da977932457ceef00f79.js:1:6685)
    at w (https://deno.land/_next/static/chunks/framework.da977932457ceef00f79.js:1:3286)
    at j (https://deno.land/_next/static/chunks/framework.da977932457ceef00f79.js:1:6918)
    at w (https://deno.land/_next/static/chunks/framework.da977932457ceef00f79.js:1:3286)
    at U (https://deno.land/_next/static/chunks/framework.da977932457ceef00f79.js:1:8346)
    at j (https://deno.land/_next/static/chunks/framework.da977932457ceef00f79.js:1:7100)
    at w (https://deno.land/_next/static/chunks/framework.da977932457ceef00f79.js:1:3286)
    at j (https://deno.land/_next/static/chunks/framework.da977932457ceef00f79.js:1:6918)
```

### PR DESCRIPTION
**Pull request by kt3k**
closes #1773

### CODE REVIEW

File: util/local_storage_utils.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2020-2021 the Deno authors. All rights reserved. MIT license.
+
+export function getItem(key: string) {
+  if (typeof window === "undefined") {
+    return null;
+  }
```
Comment (satyarohith) [inline util/local_storage_utils.ts:6]:
```suggestion
```

I guess this is not required as the below try block will throw when accessing a property of undefined?

File: util/local_storage_utils.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    return window.localStorage.getItem(key);
+  } catch (_e) {
+    return null;
+  }
+}
+
+export function setItem(key: string, value: string) {
+  if (typeof window === "undefined") {
+    return;
+  }
```
Comment (satyarohith) [inline util/local_storage_utils.ts:17]:
```suggestion
```

The reasoning is explained in the previous comment.

File: util/local_storage_utils.ts (hunk 1 above)
Comment (kt3k) [reply util/local_storage_utils.ts:6]: makes sense

**Review by satyarohith — APPROVED** [review]
I have some feelings that we can inline the `getItem` and `setItem` code instead of having a separate file for them. But I'm not sure. So I'll leave it up to you.

LGTM.

**Comment by kt3k** [comment]
@satyarohith Thank you for your review. I prefer to keep these utils because in my view this error handling is not good in the sense it just ignores every error. So keeping this util prevents this abnormal handling spread across the source code.

**Pull request merged**